xo-game/src/Components/Board/Board.jsx

```diff
@@ -1,22 +1,38 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useState } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
 import Xicon from "../icons/Xicon";
 import Oicon from "../icons/Oicon";
 import "./board.css";
 import BoardCard from "./BoardCard";
+import RestartPoupup from "../BoardPoupup/RestartPoupup";
 
 const Board = () => {
   const location = useLocation();
   const navigate = useNavigate();
   const players = location.state?.players || [];
-  const squares = ["", "O", "X", "", "O", "X", "", "", ""];
+  const [squares, setSquares] = useState(new Array(9).fill(""));
+  const [isRestartPopupOpen, setIsRestartPopupOpen] = useState(false);
+  const [xnext, setXnext] = useState(true);
```

Comment:
xnext flag is controlling the players turn ? 
but if you controlled the players turn in another state before, pass it here and use the same state